### PR TITLE
Feature/models and controllers

### LINKED ISSUE #4: [2pt] Define Models & Controllers
- [ ] Define Models
- [ ] Develop Controllers

### PR DESCRIPTION
**Pull request by ClaudiaRojasSoto**
### Description
This Pull Request aims to add the functionality of models and controllers to fulfill the specified requirements. Login or authentication logic is not included, as only the username is required to access the website.

### Changes Made
- [ ] Models have been defined for "Users", "Cars" and "Reservations" in accordance with the requirements.
- [ ] Controllers have been developed to handle operations such as "add," "delete," and others related to the selected items.

### Additional Features (for teams of 3-4 and 5 members)
- [ ] The endpoint to "Delete item" functionality has been added, allowing users to view a list of all items with a "Delete" button.
- [ ] Deleted items are marked as removed and are no longer displayed in the main list.
- [ ] Appointment booking has been implemented, allowing users to select a date and city (username and selected item are auto-filled).
- [ ] The "My Reservations" functionality has been added, displaying a list of the user's reservations with information about the item name, date, and city.

### Notes
Please carefully review the changes made to ensure they meet the aforementioned requirements.

### Assigned Reviewers
@PabloBona 
@alexmedinasf 

### Additional Comments
Any comments or feedback are welcome. Thank you for your review!

Greetings,
@ClaudiaRojasSoto

### CODE REVIEW

File: app/controllers/api/cars_controller.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+class Api::CarsController < ApplicationController
+  skip_before_action :set_current_user, only: [:create]
+  def index
+    @reservations = current_user.reservations.includes(:car).where(cars: { removed: false })
```
Comment (PabloBona) [inline app/controllers/api/cars_controller.rb:4]: Great using includes its an efficient way to retrieve the necessary data while minimizing the database load.

**Review by PabloBona — APPROVED** [review]
# Hi @ClaudiaRojasSoto 
Approved ✔
Great job, Your project is complete! 
There is nothing else to say other than... it's time to merge it :shipit:

Congratulations! 🎉

## Highlights
- [ ]  Descriptive PR⭐
- [ ] All required are applied ✨
- [ ]  Kudos for working in the API

**Comment by ClaudiaRojasSoto** [comment]
Thank you @PabloBona

File: app/controllers/api/cars_controller.rb (hunk 1 above)
Comment (ClaudiaRojasSoto) [reply app/controllers/api/cars_controller.rb:4]: It was very hard, but... Thank you!

**Pull request merged**